src/components/DateRangePicker.js

```diff
@@ -0,0 +1,462 @@
+const React = require('react');
+const Radium = require('radium');
+const moment = require('moment');
+
+const Icon = require('./Icon');
+
+const StyleConstants = require('../constants/Style');
+
+const DatePicker = React.createClass({
+  propTypes: {
+    format: React.PropTypes.string,
+    locale: React.PropTypes.string,
+    minimumDate: React.PropTypes.number,
+    onDateSelect: React.PropTypes.func,
+    placeholderText: React.PropTypes.string,
+    primaryColor: React.PropTypes.string,
+    selectedEndDate: React.PropTypes.number,
+    selectedStartDate: React.PropTypes.number,
+    style: React.PropTypes.object
+  },
+
+  getDefaultProps () {
+    return {
+      format: 'MMM D, YYYY',
+      locale: 'en',
+      onDateSelect () {},
+      placeholderText: 'Select A Date Range',
+      primaryColor: StyleConstants.Colors.PRIMARY
+    };
+  },
+
+  getInitialState () {
+    return {
+      currentDate: this.props.selectedStartDate || moment().unix(),
+      showCalendar: false
+    };
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.selectedStartDate && newProps.selectedStartDate !== this.props.selectedStartDate) {
+      this.setState({
+        currentDate: newProps.selectedStartDate
+      });
+    }
+  },
+
+  _handleDateSelect (date) {
+    const selectedStartDate = this.props.selectedStartDate;
+    const selectedEndDate = this.props.selectedEndDate;
+    let updatedDates = [null, null];
+
+    updatedDates = this._deselectDate(date, selectedStartDate, selectedEndDate);
+    updatedDates = updatedDates || this._startAndEndDate(date, selectedStartDate, selectedEndDate);
```

Comment:
On first read through I assumed `_startAndEndDate` was called if there were start and end dates because of the name. I see now it is called if `updatedDates` is falsey and then handles the situation where there are start and end dates.